### ISSUE TITLE
"Did not find any type information for expression" on strict equality comparisons of prop values in WHERE clauses

### ISSUE THREAD

**Issue by drew-moore** [neo4j/neo4j#12114]
**Neo4j Version:** 3.5.0
**Operating System:** Ubuntu 18.04  
**API:** Cypher via neo4j-browser (v 3.2.13)

### Steps to reproduce

run: 

```
CREATE (:foo{value:1, id:"abc"}), 
	   (:foo{value:2, id:"def"}), 
	   (:foo{value:2, id:"ghi"}), 
	   (:foo{value:3, id:"jkl"})
```

run: 
```
MATCH (cursor:foo{id:"def"}) WITH cursor 
MATCH (a:foo) 
WHERE a.value > cursor.value OR (a.value = cursor.value AND a.id >= cursor.id)
RETURN a ORDER BY a.value ASC, a.id ASC
```

see resulting error: 
```
Neo.DatabaseError.Statement.ExecutionFailed: Did not find any type information for expression Property(Variable(cursor),PropertyKeyName(value))
```

This problem does **not** occur when replacing the `WHERE` clause by either: 

```	       
WHERE a.value > cursor.value OR (a.value >= cursor.value AND a.id >= cursor.id) 
```

(replacing `=` by `>=`), or 
```
WHERE a.value = cursor.value
```

The same error occurs in #12100, also involving a strict equality comparison of node prop values. That one was attributed to a driver version mistmatch, but I assume that's not the case here since we're reproducing it in the browser.

**Comment by firebovine** [comment]
We're also encountering this issue. Weirdly, we found:

```
WHERE
  (a.value >= b.value AND a.value <= b.value) OR (a.value >= c.value AND a.value <= c.value)
```

works fine,  which is logically equivalent to our original statement:

```
WHERE 
  a.value = b.value OR a.value = c.value
```

This original statement being equivalent to the above bug, and triggers the same exceptions.

**Comment by richcoles** [comment]
**Neo4j Version**: 3.5.0
**API:** Cypher via neo4j-browser (v 3.2.13)

We also see this issue and the same error message when one property in the exact comparison does not exist (a.otherProperty)

```
WHERE 
  (a.otherProperty = cursor.value OR a.value = cursor.value)
```

This works in 3.4.0.

**Comment by Hunterness** [comment]
Hi all, thanks for reporting, we will look into this.

**Comment by Lojjs** [comment]
@drew-moore, @firebovine and @richcoles This is a known bug which has recently been fixed. The fix will be included in the soon to come 3.5.2 release.

Best regards Louise, Neo4j Cypher team
